File: internal/router/analytics.go
package router

import "github.com/gin-gonic/gin"

func registerAnalyticsRoutes(rg *gin.RouterGroup, h *Handlers) {
	analytics := rg.Group("/analytics")
	{
		analytics.GET("/economic", h.Analytics.GetEconomicReport)
		analytics.GET("/efficiency", h.Analytics.GetEfficiencyReport)
		analytics.GET("/reports", h.Analytics.GetRunReport)
		analytics.GET("/compare", h.Analytics.GetComparison)
		analytics.GET("/export", h.Analytics.ExportReport)
		analytics.GET("/timeseries", h.Analytics.GetTimeSeries)
		analytics.GET("/predictions/efficiency", h.Prediction.PredictEfficiency)
		analytics.GET("/predictions/fault", h.Prediction.PredictFault)
		analytics.GET("/predictions/spare-parts", h.Prediction.ForecastSpareParts)
		analytics.GET("/predictions/maintenance-window", h.Prediction.RecommendMaintenanceWindow)
		analytics.POST("/predictions/:id/workorder", h.Prediction.GenerateWorkorder)
		analytics.GET("/strategy/optimize", h.Prediction.OptimizeStrategy)
		analytics.GET("/health", h.Prediction.GetHealthOverview)
		analytics.GET("/health/station", h.Prediction.GetHealthByStation)
		analytics.GET("/health/trend", h.Prediction.GetHealthTrend)
	}
}

File: frontend/src/api/analytics.ts
import client from './client';

export const analyticsApi = {
  economic: (params?: Record<string, unknown>) =>
    client.get<unknown>('/analytics/economic', { params }),

  efficiency: (params?: Record<string, unknown>) =>
    client.get<unknown>('/analytics/efficiency', { params }),

  reports: (params?: Record<string, unknown>) =>
    client.get<unknown>('/analytics/reports', { params }),

  compare: (params?: Record<string, unknown>) =>
    client.get<unknown>('/analytics/compare', { params }),

  timeseries: (params?: Record<string, unknown>) =>
    client.get<unknown>('/analytics/timeseries', { params }),

  export: (params?: Record<string, unknown>) =>
    client.get('/analytics/export', { params, responseType: 'blob' }),

  predictions: {
    efficiency: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/predictions/efficiency', { params }),
    fault: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/predictions/fault', { params }),
    strategy: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/strategy/optimize', { params }),
    spareParts: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/predictions/spare-parts', { params }),
    maintenanceWindow: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/predictions/maintenance-window', { params }),
    generateWorkorder: (predictionId: string) =>
      client.post<unknown>(`/analytics/predictions/${predictionId}/workorder`),
  },

  health: {
    overview: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/health', { params }),
    byStation: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/health/station', { params }),
    trend: (params?: Record<string, unknown>) =>
      client.get<unknown>('/analytics/health/trend', { params }),
  },
};
